validation-test/stdlib/Algorithm.swift

```diff
@@ -24,15 +24,31 @@ public func == (
 }
 
 // FIXME(prext): move this struct to the point of use.
-Algorithm.test("min,max") {
-  expectEqual(2, min(3, 2))
-  expectEqual(3, min(3, 7, 5))
-  expectEqual(3, max(3, 2))
-  expectEqual(7, max(3, 7, 5))
-
-  // FIXME: add tests that check that min/max return the
-  // first element of the sequence (by reference equality) that satisfy the
-  // condition.
+Algorithm.test("Min/Max") {
+  expectEqual(0, min(0, 1))
+  expectEqual(0, min(1, 0))
```

Comment:
Sorry, what I had in mind was more like:

```
// Identities are unique in this set.
let a1 = MinimalComparableValue(0, identity: 1)
let a2 = MinimalComparableValue(0, identity: 2)
let a3 = MinimalComparableValue(0, identity: 3)
let b1 = MinimalComparableValue(1, identity: 4)
let b2 = MinimalComparableValue(1, identity: 5)
let b3 = MinimalComparableValue(1, identity: 6)
let c1 = MinimalComparableValue(2, identity: 7)

// 2-arg min().
expectEqual(a1.identity, min(a1, b1).identity)
expectEqual(a1.identity, min(b1, a1).identity)
expectEqual(a1.identity, min(a1, a2).identity)

// 3+-arg min().
// a1, b1, c1
// b1, a1, c1
// c1, b1, a1
// c1, a1, b1
// a1, a2, a3
// a1, a2, b1
// a1, b1, a2
// b1, a1, a2

// a1, b1, a2, b2
// b1, a1, a2, b2
// c1, b1, b2, a1
// c1, b1, a1, a2

// 2-arg max().
// ...

// 3+-arg max().
// ...
```

That is, you don't need to even check values (nor have separate tests for them) if you assign unique identities.